lib/open_food_network/products_renderer.rb

```diff
@@ -54,14 +54,16 @@ def taxon_order
     end
 
     def all_variants_for_shop
-      # We use the in_stock? method here instead of the in_stock scope because we need to
-      # look up the stock as overridden by VariantOverrides, and the scope method is not affected
-      # by them.
-      scoper = OpenFoodNetwork::ScopeVariantToHub.new(@distributor)
-      Spree::Variant.
-        for_distribution(@order_cycle, @distributor).
-        each { |v| scoper.scope(v) }.
-        select(&:in_stock?)
+      @all_variants_for_shop ||= begin
+                                   # We use the in_stock? method here instead of the in_stock scope
+                                   # because we need to look up the stock as overridden by
+                                   # VariantOverrides, and the scope method is not affected by them.
+                                   scoper = OpenFoodNetwork::ScopeVariantToHub.new(@distributor)
+                                   @all_variants_for_shop = Spree::Variant.
```

Comment:
`@all_variants_for_shop =` is redundant here

Reply:
Ah I missed that. Yeah, that shouldn't be there.

Reply:
Whoops! Sorry, that was from an earlier change. Changed already.